// ice/src/Ice/Endpoint.js
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************
//
// Ice version 3.6.3
//
// <auto-generated>
//
// Generated from file `Endpoint.ice'
//
// Warning: do not edit this file.
//
// </auto-generated>
//

/* slice2js browser-bundle-skip */
(function(module, require, exports)
{
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */
    var __M = require("../Ice/ModuleRegistry").Ice.__M;
    var Ice = __M.require(module, 
    [
        "../Ice/Object",
        "../Ice/ObjectPrx",
        "../Ice/Long",
        "../Ice/HashMap",
        "../Ice/HashUtil",
        "../Ice/ArrayUtil",
        "../Ice/StreamHelpers",
        "../Ice/Version",
        "../Ice/BuiltinSequences",
        "../Ice/EndpointF"
    ]).Ice;
    
    var Slice = Ice.Slice;
/* slice2js browser-bundle-skip-end */

    Object.defineProperty(Ice, 'TCPEndpointType', {
        value: 1
    });

    Object.defineProperty(Ice, 'SSLEndpointType', {
        value: 2
    });

    Object.defineProperty(Ice, 'UDPEndpointType', {
        value: 3
    });

    Object.defineProperty(Ice, 'WSEndpointType', {
        value: 4
    });

    Object.defineProperty(Ice, 'WSSEndpointType', {
        value: 5
    });

    /**
     * Base class providing access to the endpoint details.
     * 
     **/
    Ice.EndpointInfo = Slice.defineLocalObject(
        function(timeout, compress)
        {
            this.timeout = timeout !== undefined ? timeout : 0;
            this.compress = compress !== undefined ? compress : false;
        });

    /**
     * The user-level interface to an endpoint.
     * 
     **/
    Ice.Endpoint = Slice.defineLocalObject();

    /**
     * Provides access to the address details of a IP endpoint.
     * 
     * @see Endpoint
     * 
     **/
    Ice.IPEndpointInfo = Slice.defineLocalObject(
        function(timeout, compress, host, port, sourceAddress)
        {
            Ice.EndpointInfo.call(this, timeout, compress);
            this.host = host !== undefined ? host : "";
            this.port = port !== undefined ? port : 0;
            this.sourceAddress = sourceAddress !== undefined ? sourceAddress : "";
        },
        Ice.EndpointInfo);

    /**
     * Provides access to a TCP endpoint information.
     * 
     * @see Endpoint
     * 
     **/
    Ice.TCPEndpointInfo = Slice.defineLocalObject(
        function(timeout, compress, host, port, sourceAddress)
        {
            Ice.IPEndpointInfo.call(this, timeout, compress, host, port, sourceAddress);
        },
        Ice.IPEndpointInfo);

    /**
     * Provides access to an UDP endpoint information.
     * 
     * @see Endpoint
     * 
     **/
    Ice.UDPEndpointInfo = Slice.defineLocalObject(
        function(timeout, compress, host, port, sourceAddress, mcastInterface, mcastTtl)
        {
            Ice.IPEndpointInfo.call(this, timeout, compress, host, port, sourceAddress);
            this.mcastInterface = mcastInterface !== undefined ? mcastInterface : "";
            this.mcastTtl = mcastTtl !== undefined ? mcastTtl : 0;
        },
        Ice.IPEndpointInfo);

    /**
     * Provides access to a WebSocket endpoint information.
     * 
     **/
    Ice.WSEndpointInfo = Slice.defineLocalObject(
        function(timeout, compress, host, port, sourceAddress, resource)
        {
            Ice.TCPEndpointInfo.call(this, timeout, compress, host, port, sourceAddress);
            this.resource = resource !== undefined ? resource : "";
        },
        Ice.TCPEndpointInfo);

    /**
     * Provides access to the details of an opaque endpoint.
     * 
     * @see Endpoint
     * 
     **/
    Ice.OpaqueEndpointInfo = Slice.defineLocalObject(
        function(timeout, compress, rawEncoding, rawBytes)
        {
            Ice.EndpointInfo.call(this, timeout, compress);
            this.rawEncoding = rawEncoding !== undefined ? rawEncoding : new Ice.EncodingVersion();
            this.rawBytes = rawBytes !== undefined ? rawBytes : null;
        },
        Ice.EndpointInfo);
/* slice2js browser-bundle-skip */
    exports.Ice = Ice;
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */
}
(typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? module : undefined,
 typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? require : this.Ice.__require,
 typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? exports : this));
/* slice2js browser-bundle-skip-end */
