// ice/src/Glacier2/SSLInfo.js
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************
//
// Ice version 3.6.3
//
// <auto-generated>
//
// Generated from file `SSLInfo.ice'
//
// Warning: do not edit this file.
//
// </auto-generated>
//

/* slice2js browser-bundle-skip */
(function(module, require, exports)
{
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */
    var __M = require("../Ice/ModuleRegistry").Ice.__M;
    var Ice = __M.require(module, 
    [
        "../Ice/Struct",
        "../Ice/Long",
        "../Ice/HashMap",
        "../Ice/HashUtil",
        "../Ice/ArrayUtil",
        "../Ice/StreamHelpers",
        "../Ice/BuiltinSequences"
    ]).Ice;
    
    var Slice = Ice.Slice;
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */

    var Glacier2 = __M.module("Glacier2");
/* slice2js browser-bundle-skip-end */

    /**
     * Information taken from an SSL connection used for permissions
     * verification.
     * 
     * @see PermissionsVerifier
     * 
     **/
    Glacier2.SSLInfo = Slice.defineStruct(
        function(remoteHost, remotePort, localHost, localPort, cipher, certs)
        {
            this.remoteHost = remoteHost !== undefined ? remoteHost : "";
            this.remotePort = remotePort !== undefined ? remotePort : 0;
            this.localHost = localHost !== undefined ? localHost : "";
            this.localPort = localPort !== undefined ? localPort : 0;
            this.cipher = cipher !== undefined ? cipher : "";
            this.certs = certs !== undefined ? certs : null;
        },
        true,
        function(__os)
        {
            __os.writeString(this.remoteHost);
            __os.writeInt(this.remotePort);
            __os.writeString(this.localHost);
            __os.writeInt(this.localPort);
            __os.writeString(this.cipher);
            Ice.StringSeqHelper.write(__os, this.certs);
        },
        function(__is)
        {
            this.remoteHost = __is.readString();
            this.remotePort = __is.readInt();
            this.localHost = __is.readString();
            this.localPort = __is.readInt();
            this.cipher = __is.readString();
            this.certs = Ice.StringSeqHelper.read(__is);
        },
        12, 
        false);
/* slice2js browser-bundle-skip */
    exports.Glacier2 = Glacier2;
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */
}
(typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? module : undefined,
 typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? require : this.Ice.__require,
 typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? exports : this));
/* slice2js browser-bundle-skip-end */
