// ice/src/IceStorm/Metrics.js
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************
//
// Ice version 3.6.3
//
// <auto-generated>
//
// Generated from file `Metrics.ice'
//
// Warning: do not edit this file.
//
// </auto-generated>
//

/* slice2js browser-bundle-skip */
(function(module, require, exports)
{
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */
    var __M = require("../Ice/ModuleRegistry").Ice.__M;
    var Ice = __M.require(module, 
    [
        "../Ice/Object",
        "../Ice/ObjectPrx",
        "../Ice/Long",
        "../Ice/HashMap",
        "../Ice/HashUtil",
        "../Ice/ArrayUtil",
        "../Ice/StreamHelpers"
    ]).Ice;
    
    var IceMX = require("../Ice/Metrics").IceMX;
    var Slice = Ice.Slice;
/* slice2js browser-bundle-skip-end */

    /**
     * Provides information on IceStorm topics.
     * 
     **/
    IceMX.TopicMetrics = Slice.defineObject(
        function(id, total, current, totalLifetime, failures, published, forwarded)
        {
            IceMX.Metrics.call(this, id, total, current, totalLifetime, failures);
            this.published = published !== undefined ? published : new Ice.Long(0, 0);
            this.forwarded = forwarded !== undefined ? forwarded : new Ice.Long(0, 0);
        },
        IceMX.Metrics, undefined, 2,
        [
            "::Ice::Object",
            "::IceMX::Metrics",
            "::IceMX::TopicMetrics"
        ],
        -1,
        function(__os)
        {
            __os.writeLong(this.published);
            __os.writeLong(this.forwarded);
        },
        function(__is)
        {
            this.published = __is.readLong();
            this.forwarded = __is.readLong();
        },
        false);

    IceMX.TopicMetricsPrx = Slice.defineProxy(IceMX.MetricsPrx, IceMX.TopicMetrics.ice_staticId, undefined);

    Slice.defineOperations(IceMX.TopicMetrics, IceMX.TopicMetricsPrx);

    /**
     * Provides information on IceStorm subscribers.
     * 
     **/
    IceMX.SubscriberMetrics = Slice.defineObject(
        function(id, total, current, totalLifetime, failures, queued, outstanding, delivered)
        {
            IceMX.Metrics.call(this, id, total, current, totalLifetime, failures);
            this.queued = queued !== undefined ? queued : 0;
            this.outstanding = outstanding !== undefined ? outstanding : 0;
            this.delivered = delivered !== undefined ? delivered : new Ice.Long(0, 0);
        },
        IceMX.Metrics, undefined, 2,
        [
            "::Ice::Object",
            "::IceMX::Metrics",
            "::IceMX::SubscriberMetrics"
        ],
        -1,
        function(__os)
        {
            __os.writeInt(this.queued);
            __os.writeInt(this.outstanding);
            __os.writeLong(this.delivered);
        },
        function(__is)
        {
            this.queued = __is.readInt();
            this.outstanding = __is.readInt();
            this.delivered = __is.readLong();
        },
        false);

    IceMX.SubscriberMetricsPrx = Slice.defineProxy(IceMX.MetricsPrx, IceMX.SubscriberMetrics.ice_staticId, undefined);

    Slice.defineOperations(IceMX.SubscriberMetrics, IceMX.SubscriberMetricsPrx);
/* slice2js browser-bundle-skip */
    exports.IceMX = IceMX;
/* slice2js browser-bundle-skip-end */
/* slice2js browser-bundle-skip */
}
(typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? module : undefined,
 typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? require : this.Ice.__require,
 typeof(global) !== "undefined" && typeof(global.process) !== "undefined" ? exports : this));
/* slice2js browser-bundle-skip-end */
